index.js

```diff
@@ -241,26 +264,26 @@ const ContainerService = {
       return Promise.resolve(null);
     }
 
-    return new Promise(resolve => {
-      this._containerTabIterator(args.userContextId, tab => {
-        const object = this._createTabObject(tab);
+    const tabsToClose = [];
 
-        // This tab is going to be closed. Let's mark this tabObject as
-        // non-active.
-        object.active = false;
+    this._containerTabIterator(args.userContextId, tab => {
+      const object = this._createTabObject(tab);
 
-        getFavicon(object.url).then(url => {
-          object.favicon = url;
-        }).catch(() => {
-          object.favicon = "";
-        });
+      // This tab is going to be closed. Let's mark this tabObject as
+      // non-active.
+      object.active = false;
 
-        this._identitiesState[args.userContextId].hiddenTabUrls.push(object);
-        tab.close();
+      getFavicon(object.url).then(url => {
+        object.favicon = url;
+      }).catch(() => {
+        object.favicon = "";
```

Comment:
You are right, but in theory, from an index.js point of view, getTabs cannot be called yet.
Plus, it seems that getFavicon is almost noop if the icon is already known.